File: editor/src/de/fatox/meta/modules/MetaUIModule.java
package de.fatox.meta.modules;

import de.fatox.meta.Meta;
import de.fatox.meta.api.dao.MetaData;
import de.fatox.meta.injection.Provides;
import de.fatox.meta.injection.Singleton;
import de.fatox.meta.ui.MetaEditorUI;

/**
 * Created by Frotty on 07.06.2016.
 */
public class MetaUIModule {
    public MetaUIModule() {
        Meta.inject(this);
    }

    @Provides
    @Singleton
    public MetaData metaData() {
        return new MetaData();
    }

    @Provides
    @Singleton
    public MetaEditorUI metaEditorUI() {
        return new MetaEditorUI();
    }

}

File: editor/src/main/kotlin/de/fatox/meta/modules/MetaUIModule.kt
package de.fatox.meta.modules

import com.badlogic.gdx.Gdx
import de.fatox.meta.api.lang.LanguageBundle
import de.fatox.meta.assets.MetaData
import de.fatox.meta.injection.MetaInject
import de.fatox.meta.lang.MetaLanguageBundle
import de.fatox.meta.ui.MetaEditorUI

object MetaUIModule {
	/** Forces this module's registration block to run before editor services are resolved. */
	internal fun initialize() = Unit

	init {
		MetaInject.global {
			singleton("meta", "gameName")
			singleton { MetaData() }
			singleton { MetaEditorUI() }
			singleton<LanguageBundle> { MetaLanguageBundle(Gdx.files.internal("lang/MetagineBundle")) }
		}
	}
}
